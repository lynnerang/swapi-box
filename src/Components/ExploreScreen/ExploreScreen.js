import React from 'react';
import './ExploreScreen.scss';
import { NavLink } from 'react-router-dom';

function ExploreScreen(props) {
  return (
    <section className='category-area'>
      <h2 className='category-header'>CHOOSE A CATEGORY</h2>
      <div className='category-options'>
        <NavLink className='category-card' to='/people' onClick={props.getPeopleData}>
          <img className='category-img' src={require('../../images/peopleicon.png')}aria-hidden='true' alt='darth vader icon'/>
          <p className='category-text'>People</p>
        </NavLink>
        <NavLink className='category-card' to='/planets' onClick={props.getPlanetData}>
          <img className='category-img' src={require('../../images/planeticon.png')}aria-hidden='true' alt='planet icon' />
          <p className='category-text'>Planets</p>
        </NavLink>
        <NavLink className='category-card' to='/vehicles'>
          <img className='category-img' src={require('../../images/vehiclesicon.png')} aria-hidden='true' alt='millenium falcon icon'/>
          <p className='category-text'>Vehicles</p>
        </NavLink>
      </div>
    </section>
  );
}

export default ExploreScreen;
