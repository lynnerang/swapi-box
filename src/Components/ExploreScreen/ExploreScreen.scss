.category-area {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding-top: 20px;
}

.category-header {
  color: white;
  margin-bottom: 25px;
  opacity: .7;
}

.category-options {
  display: flex;
  align-items: center;
}

.category-img {
  height: 80px;
  width: 80px;
  color: white;
}

.category-text {
  color: white;
  font-size: 26px;
  margin: 0;
  margin-top: 10px;
}

.category-card {
  height: 250px;
  width: 250px;
  border: 1px solid white;
  border-radius: 10px;
  opacity: .7;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 30px 60px;
  text-decoration: none;

  &:hover {
    opacity: 1;

    .category-img,
    .category-text {
      opacity: 1;
    }
  }
}



