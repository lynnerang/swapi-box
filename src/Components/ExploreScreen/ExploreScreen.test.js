import React from 'react';
import ExploreScreen from './ExploreScreen';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() })

describe('ExploreScreen', () => {
  let wrapper; 

  beforeEach(() => {
    wrapper = shallow( <ExploreScreen /> )
  })

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});