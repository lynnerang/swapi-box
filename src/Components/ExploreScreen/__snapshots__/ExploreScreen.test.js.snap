// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreScreen Should match the snapshot 1`] = `ShallowWrapper {}`;
